internal/util/cluster_test.go

```diff
@@ -114,3 +117,60 @@ func TestGenerateNodeAffinity(t *testing.T) {
 		}
 	})
 }
+
+func TestValidateLabels(t *testing.T) {
+	t.Run("valid", func(t *testing.T) {
+		inputs := []map[string]string{
+			{"key": "value"},
+			{"example.com/key": "value"},
+			{"key1": "value1", "key2": "value2"},
+		}
+
+		for _, input := range inputs {
+			t.Run(formatLabelStr(input), func(*testing.T) {
+				err := ValidateLabels(input)
+
+				if err != nil {
+					t.Fatalf("expected no error, got: %s", err.Error())
+				}
+			})
+		}
+	})
+
+	t.Run("invalid", func(t *testing.T) {
+		inputs := []map[string]string{
+			{"key=value": "value"},
+			{"key": "value", "": ""},
+			{"b@d": "value"},
+			{"b@d-prefix/key": "value"},
+			{"really/bad/prefix/key": "value"},
+			{"key": "v\\alue"},
+		}
+
+		for _, input := range inputs {
+			t.Run(formatLabelStr(input), func(t *testing.T) {
+				err := ValidateLabels(input)
+
+				if err == nil {
+					t.Fatalf("expected an invalid input error.")
+				}
+
+				if !errors.Is(err, ErrLabelInvalid) {
+					t.Fatalf("expected an ErrLabelInvalid error.")
+				}
```

Comment:
These two can be combined. It's good to mention the failing value in this case.

```suggestion
				if !errors.Is(err, ErrLabelInvalid) {
					t.Fatalf("expected an ErrLabelInvalid error, got %T: %v", err, err)
				}
```